libfoundation/src/foundation-handler.cpp

```diff
@@ -90,10 +93,141 @@ const MCHandlerCallbacks *MCHandlerGetCallbacks(MCHandlerRef self)
 
 ////////////////////////////////////////////////////////////////////////////////
 
+static void __exec_closure(ffi_cif *cif, void *ret, void **args, void *user_data)
+{
+    MCHandlerRef t_handler;
+    t_handler = (MCHandlerRef)user_data;
+    
+    MCTypeInfoRef t_signature;
+    t_signature = t_handler -> typeinfo;
+    
+    uindex_t t_arity;
+    t_arity = MCHandlerTypeInfoGetParameterCount(t_signature);
+    
+    MCValueRef t_value_result;
+    MCValueRef t_value_args[16];
+    uindex_t t_arg_index;
+    t_arg_index = 0;
+    t_value_result = nil;
+    for(t_arg_index = 0; t_arg_index < t_arity; t_arg_index++)
+    {
+        MCHandlerTypeFieldMode t_mode;
+        t_mode = MCHandlerTypeInfoGetParameterMode(t_signature, t_arg_index);
+        
+        MCTypeInfoRef t_type;
+        t_type = MCHandlerTypeInfoGetParameterType(t_signature, t_arg_index);
+        
+        if (t_mode != kMCHandlerTypeFieldModeIn)
+            abort();
+        
+        MCResolvedTypeInfo t_resolved_type;
+        if (!MCTypeInfoResolve(t_type, t_resolved_type))
+        {
+            MCErrorThrowGeneric(nil);
```

Comment:
I'll fix the missing error message by making it an 'unimplemented' error.

I've refactored the UnboundTypeError message from libscript into libfoundation and added a throw method for it. I'm not sure MCTypeInfoResolve should necessarily throw - there might be current contexts where that wouldn't be correct.

To be fair, the type resolution needs a little bit of a rethink. There should be no need to resolve the types really as everything should be patched up as extensions load (but that isn't the case at the moment).